src/Reporting/src/App.Metrics.Formatting.StatsD/MetricsStatsDStringOutputFormatter.cs

```diff
@@ -63,5 +65,18 @@ public MetricsStatsDStringOutputFormatter(MetricsStatsDOptions options)
             await using var writer = new MetricSnapshotStatsDStringWriter(output, _samplers, _options);
             serializer.Serialize(writer, metricsData, MetricFields);
         }
+
+        public Task<List<string>> WriteAsync(MetricsDataValueSource metricsData, CancellationToken cancellationToken = default)
+        {
+            var writer = new MetricSnapshotStatsDStringWriter(null, _samplers, _options);
```

Comment:
are these both single use objects? Or can they be re-used across multiple successive calls?

Reply:
Its single use, because of the sampler instance inside of the class, also the reference to the stream object for the standard unchunked write mode